examples/send_and_subscribe.rs

```diff
@@ -0,0 +1,153 @@
+extern crate env_logger;
+extern crate stomp;
+extern crate tokio;
+#[macro_use]
+extern crate futures;
+
+use futures::future::join_all;
+use stomp::connection::{Credentials, HeartBeat};
+use stomp::frame::Frame;
+use stomp::header::{Destination, Header, SuppressedHeader};
+use stomp::session::{Session, SessionEvent};
+use stomp::session_builder::SessionBuilder;
+use stomp::subscription::AckMode;
+use tokio::prelude::*;
+
+struct ExampleSession {
+    session: Session,
+    destination: String,
+    session_number: u32,
+
+    subscription_id: Option<String>,
+}
+
+impl ExampleSession {
+    fn on_connected(&mut self) {
+        println!("Example session established.");
+        let destination = &self.destination;
+        println!("Subscribing to '{}'.", destination);
+
+        let subscription_id = self
+            .session
+            .subscription(destination)
+            .with(AckMode::Auto)
+            .with(Header::new("custom-subscription-header", "lozenge"))
+            .start();
+
+        self.subscription_id = Some(subscription_id);
+
+        self.session.message(destination, "Animal").send();
+        self.session.message(destination, "Vegetable").send();
+        self.session.message(destination, "Mineral").send();
+    }
+
+    fn on_gilbert_and_sullivan_reference(&mut self, frame: Frame) {
+        println!(
+            "Another droll reference!: '{}'",
+            std::str::from_utf8(&frame.body).expect("Non-utf8 bytes")
+        );
+    }
+}
+
+impl Future for ExampleSession {
```

Comment:
This `stomp` library has already changed its underlying implementation quite a few times. It's been singlethreaded, multithreaded, mio-based, `tokio-core`-based, and now `tokio`-based. It's not stable yet, so we can still make changes to the APIs. However, most of the changes in each migration have been implementation details. It's best if we don't require the end users of the API to change their code unless they have to.

In this example, the end user should only have to implement the methods on the `Handler` trait that they're interested in. If there's new, `tokio`-specific code that needs to be written for the Handler trait to work properly (for example, implementing `Future`), that should be an implementation detail handled by the library itself.

Reply:
`ReceiptHandler`/`Handler` is removed before I started to work on the library.

https://github.com/zslayton/stomp-rs/commit/fb62a6ece2910048bebddd408cdf2464542c4576#diff-4c9ae8d0d599659f5476e14edb82cbaeL72